pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -64,6 +66,15 @@ type PipelineTaskParam struct {
 	Value string `json:"value"`
 }
 
+// ClusterBinding is used to bind a cluster from a PipelineParams to a cluster required
+// as an input for a task.
+type ClusterBinding struct {
+	// ClusterName is the string that the PipelineParams will use to identify this source.
+	ClusterName string `json:"clusterName"`
+	// InputName is the string the Task will use to identify this cluster in its inputs.
+	InputName string `json:"inputName"`
```

Comment:
Lets get `ClusterBinding` and `SourceBinding` structs fields in sync.
Shd we rename `ClusterName` to `Key` and also drop `SourceKey` to `Key` ?
Can we consider renamung `ClusterBinding.InputName` and `SourceBinding.Name` to something like `BoundTo`.
`SourceBinding.Key` and `ClusterBindings.Key` avoids stuttering. 
```
        clusterBindings:
          -Key: test
```

We can also define a struct `Binding` and then include common fields in those.

Reply:
> Shd we rename ClusterName to Key and also drop SourceKey to Key ?

I agree with you to stay in sync across these bindings to make the pipeline yaml more readable. I like `key` for bindings. 👍 

> Can we consider renaming ClusterBinding.InputName and SourceBinding.Name to something like `BoundTo`

I think keyword `bindings` in `ClusterBindings`/ `SourceBindings` convey that this is templated to an another definition(like task). I think `BoundTo` might be redundant.

How about explicitly calling out `taskInputName`(since both are bound to tasks) or keep the original key `inputName` (more generic)?

Reply:
agreed `key` is a better name the key field
 * `ClusterBinding.InputName` becomes `ClusterBinding.Key`  
 * `SourceBinding.SourceKey` becomes `SourceBinding.Key` _(think this should be a separate PR since this PR here is about adding `ClusterBinding`)_

But since we already have the type name including the word `Binding` I feel the field `name` should be enough to clarify what it is 

eg:
```
apiVersion: pipeline.knative.dev/v1beta1
kind: Pipeline
metadata:
  name: wizzbang-pipeline
  namespace: default
spec:
    tasks:
      - name: 'test'
        taskRef:
          name: test-wizzbang-task
        inputsourceBindings:
          - name: repoUnderTest
            key: wizzbang   <- changed to key in different PR
        clusterBindings:
          - name: test-cluster  <- name of this cluster binding
            key: gke-cluster-27    <- key to refer to
    .........
```